### ISSUE TITLE
EdgeTPU tflite models, labels off by one

### ISSUE THREAD

**Issue by peacefulhill** [snowzach/doods2#108]
Tested more or less all coral.ai EdgeTPU, which all shares the same label file.
I have the same problem with all models. The labels are off by one, and the first label is considered "unknown" which is supposed to be "person".
Example, beginning of labels:
person
bicycle
car
In the detection, bicycles becomes person, cars become bicycles and persons become unknown.
This broblem does not appear for tflite models (not EdgeTPU version)

It can be sort of worked around assuming you don't care about detecting persons. If the first row with person is deleted. It all works, persons still become unknown but all other labels work.

Example, with all labels in the file:
![image](https://github.com/snowzach/doods2/assets/7250474/8bd990f8-b678-4aa4-95c9-81e9b5d495a3)

Example, with person removed from label file:
![image](https://github.com/snowzach/doods2/assets/7250474/3132ccb5-cb43-4fc3-8ee0-31f8781c517a)

And finally the default tflite (not for EdgeTPU, person at the beginning of label also here):
![image](https://github.com/snowzach/doods2/assets/7250474/3290cce6-16e7-499d-a37e-18e6fe0fb920)

**Comment by peacefulhill** [comment]
Just after finally writing an issue, I noticed the mentioning of "labelsStartFromZero: true", but I can't seem to find where to put it, my best guess is within the model as it must be model specific, but e.g. this config does not fix my issue above:

- name: coral_lite3
  type: tflite
  modelFile: /share/doods/models/efficientdet_lite3_512_ptq_edgetpu.tflite
  labelFile: /share/doods/models/coco_labels.txt
  labelsStartFromZero: true
  hwAccel: true

**Comment by peacefulhill** [comment]
Ok, so I found a difference in how the default label file is constructed and the ones you download from coral.ai. The default labels include index numbers, e.g.
0  person
1  bicycle
2  car

If I add such index number to the label files for the EdgeTPU models, it does work. However I still consider the labelsStartFromZero: true not to be working as intended. However if you have misunderstood the indended use of this variable, fell free to close the issue. It would be really nice if the labels that you download from coral.ai for the EdgeTPU would just work, but adding index numbers is of course fairly straight forward as a workaround.

**Comment by snowzach** [comment]
Yeah, you have `labelsStartFromZero` in the right place. I just reviewed all the code and it looks right. Are you sure you didn't have the number in the file when you tried it?

**Comment by peacefulhill** [comment]
Yes, I am sure as at that testing stage I had not yet understood adding index was possible. I instead temporarily worked around the issue by removing the first row. I couldn’t find any obvious issues in the code either.
It should be possible to reproduce with cora.ai downloadable models. However I will not push for further investigation as I am happy with just adding index.
Thanks for confirming I was using ghe variable correctly.
